pkg/machinery/config/types/v1alpha1/v1alpha1_validation.go

```diff
@@ -419,14 +423,21 @@ func ValidateNetworkDevices(d *Device, bondedInterfaces map[string]string, check
 }
 
 // CheckDeviceInterface ensures that the interface has been specified.
+//nolint:gocyclo
 func CheckDeviceInterface(d *Device, bondedInterfaces map[string]string) ([]string, error) {
 	var result *multierror.Error
 
 	if d == nil {
 		return nil, fmt.Errorf("empty device")
 	}
 
-	if d.DeviceInterface == "" {
+	if d.DeviceInterface == "" && d.DeviceSelector == nil {
+		result = multierror.Append(result, fmt.Errorf("[%s], [%s]: %w", "networking.os.device.interface", "networking.os.device.deviceSelector", ErrRequiredSectionOptions))
```

Comment:
this needs a unit-test to make sure we're not going to break it over time (we have validation tests already)